packages/repository-modernizer/src/restructure-config.js

```diff
@@ -339,4 +341,173 @@ async function deleteEmptyConfigDir(oldConfigFilePath) {
+                        var val = obj[key];
+                        let str = removeUnwantedChars(
+                            key,
+                            val,
+                            osgiConfigFilePath
+                        );
+                        contentToBeWritten.push(str + ",");
+                    }
+                }
+                if (contentToBeWritten.length > 1) {
+                    let lastStr = contentToBeWritten[
+                        contentToBeWritten.length - 1
+                    ].substring(
+                        0,
+                        contentToBeWritten[contentToBeWritten.length - 1]
+                            .length - 1
+                    );
+                    contentToBeWritten.pop();
+                    contentToBeWritten.push(lastStr);
+                }
+            }
+            contentToBeWritten.push("}");
+            await util.writeDataToFileAsync(
+                formattedOsgiConfigFilepath,
+                contentToBeWritten
+            );
+        } else {
+            if (osgiConfigFilePath.endsWith(constants.CONFIG_EXTENSION)) {
+                formattedOsgiConfigFilepath =
+                    osgiConfigFilePath.substring(
+                        0,
+                        osgiConfigFilePath.lastIndexOf(
+                            constants.CONFIG_EXTENSION
+                        )
+                    ) + constants.CFG_JSON_EXTENSION;
+            } else {
+                formattedOsgiConfigFilepath =
+                    osgiConfigFilePath.substring(
+                        0,
+                        osgiConfigFilePath.lastIndexOf(constants.CFG_EXTENSION)
+                    ) + constants.CFG_JSON_EXTENSION;
+            }
+            contentToBeWritten.push("{");
+            fileContent = fileContent.filter((a) => a);
+            for (let line = 0; line < fileContent.length; line++) {
+                let configLine = fileContent[line].trim();
+                let key = configLine.substring(0, configLine.indexOf("="));
+                let val = configLine.substring(configLine.indexOf("=") + 1);
+                let str = removeUnwantedChars(key, val, osgiConfigFilePath);
+                if (key != "") {
+                    if (line == fileContent.length - 1) {
+                        contentToBeWritten.push(str);
+                    } else {
+                        contentToBeWritten.push(str + ",");
+                    }
+                }
+            }
+            contentToBeWritten.push("}");
+            await util.writeDataToFileAsync(
+                formattedOsgiConfigFilepath,
+                contentToBeWritten
+            );
+        }
+        fs.unlinkSync(osgiConfigFilePath);
+        conversionStep.addOperation(
+            new ConversionOperation(
+                commons_constants.ACTION_ADDED,
+                path.dirname(formattedOsgiConfigFilepath),
+                `Formatted OSGI config file ${path.basename(
```

Comment:
`Formatted OSGi config file '${fileName}' to '${path.basename(formattedOsgiConfigFilepath)}'.`

Reply:
changed msg